tools/objinfo.ml

```diff
@@ -107,6 +109,61 @@ let print_cmt_infos cmt =
     (match cmt.cmt_impl_shape with
     | None -> printf "(none)\n"
     | Some shape -> Format.printf "\n%a" Shape.print shape)
+  end;
+  if !index then begin
+    printf "Indexed shapes:\n";
+    List.iter (fun (item, loc) ->
+      let pp_loc fmt { Location.txt; loc } =
+        Format.fprintf fmt "%a (%a)"
+          Pprintast.longident txt Location.print_loc loc
+      in
+      match item with
+      | Resolved uid ->
+          Format.printf "@[<hov 2>%a:@ %a@]@;"
+            Shape.Uid.print uid
+            pp_loc loc
+      | Unresolved shape | Approximated shape | Missing_uid shape ->
+          Format.printf "@[<hov 2>%a:@ %a@]@;"
```

Comment:
You could give slightly more info here, seems like it could be useful.
At least in the `Missing_uid` case?